chips/sam4l/src/pm.rs

```diff
@@ -187,14 +161,56 @@ pub enum OscClock {
 /// that for systems that do not work, at fast speed, they will hang or panic
 /// after several entries into WAIT mode.
 #[derive(Copy,Clone,Debug)]
-pub enum OscStartupMode {
+pub enum OscillatorStartup {
     /// Use a fast startup. ~0.5 ms in practice.
     FastStart,
 
     /// Use a slow startup. ~8.9 ms in practice.
     SlowStart,
 }
 
+/// Which source the system clock should be generated from. These are specified
+/// as system clock source appended with the clock that it is sourced from
+/// appended with the final frequency of the system. So for example, one option
+/// is to use the DFLL sourced from the RC32K with a final frequency of 48 MHz.
+///
+/// When new options (either sources or final frequencies) are needed, they
+/// should be added to this list, and then the `setup_system_clock` function
+/// can be modified to support it. This is necessary because configurations
+/// must be changed not just with the input source but also based on the
+/// desired final frequency.
+///
+/// For options utilizing an external oscillator, the configurations for that
+/// oscillator must also be provided.
+#[derive(Copy,Clone,Debug)]
+pub enum SystemClockSource {
+    /// Use the RCSYS clock (which the system starts up on anyways). Final
+    /// system frequency will be 115 kHz. Note that while this is the default,
+    /// Tock is NOT guaranteed to work on this setting and will likely fail.
+    RcsysAt115kHz,
```

Comment:
I didn't trace it at all. However, running at 115000 Hz and then expecting your USART to run at 115200 baud seems dubious at best. I wasn't surprised it failed. I bet if you removed some stuff from init and ran the blink app it would be okay.